File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char,int> freq;
        for (auto it : s){
            freq[it]++;
        }

        priority_queue<pair<int, char>> pq;
        for (auto it : freq){
            pq.push({it.second, it.first});
        }
        
        string ans;
        while(!pq.empty()){
            int f = pq.top().first;
            char ch = pq.top().second;

            ans = ans.append(f,ch);
            pq.pop();

        }
        return ans;
        

        
    }
};

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        dic = {}

        for ch in s:
            if ch in dic:
                dic[ch] += 1
            else:
                dic[ch] = 1 
        
        sorted_order = sorted(dic.items(), key = lambda x : x[1], reverse=True)
        print(sorted_order)
        # return ""
        ans = ""
        for p in sorted_order:
            ans += p[0]*p[1]
        return ans
        # for ele in sorted_order:


